pkg/controller/utils/utils.go

```diff
@@ -50,10 +63,92 @@ const (
 )
 
 // GetConcurrentReconciles returns the number of goroutines each controller should
-// use for parallel processing of their queue. For now this is a static value of 5.
-// In future this may be read from an env var set by the operator, and driven by HiveConfig.
-func GetConcurrentReconciles() int {
-	return concurrentControllerReconciles
+// use for parallel processing of their queue. Default value, if not set in
+// hive-conntrollers-config, will be 5.
+func GetConcurrentReconciles(controllerName string, configData map[string]string) (int, error) {
+	if value, ok := configData[fmt.Sprintf("%s-goroutines", controllerName)]; ok {
+		concurrentReconciles, err := strconv.Atoi(value)
+		if err != nil {
+			return 0, err
+		}
+		return concurrentReconciles, nil
+	} else if value, ok := configData["default-goroutines"]; ok {
+		concurrentReconciles, err := strconv.Atoi(value)
+		if err != nil {
+			return 0, err
+		}
+		return concurrentReconciles, nil
+	}
+	return concurrentControllerReconciles, nil
+}
+
+// GetClientRateLimiter returns the client rate limiter for the controller
+func GetClientRateLimiter(controllerName string, configData map[string]string) (flowcontrol.RateLimiter, error) {
+	qps := rest.DefaultQPS
+	var err error
+	if qpsStr, ok := configData[fmt.Sprintf("%s-client-qps", controllerName)]; ok {
+		qpsFloat64, err := strconv.ParseFloat(qpsStr, 32)
```

Comment:
This is parsed as a float, but it was written as an int. Should this be a float in HiveConfig?

Reply:
float is not supported. check [this](https://github.com/kubernetes-sigs/controller-tools/issues/245). Besides, is there any specific use case where we would like to set qps as float?

Reply:
Yeah, right, you cannot do a float in the API. But this should still not be trying to parse as a float.

Reply:
Agreed, but I still need to convert it into a float as `rest.DefaultQPS ` is a float.

Reply:
Yep.